### PR TITLE
Add function to sample random variates and fit regression model

### LINKED ISSUE #3: Add scripts to model random variates and fit a regression model to mtcars dataset.

### PR DESCRIPTION
**Pull request by jamesmbaazam**
## Improvements

This PR introduces a script to model the `mtcars` dataset in R using regression analysis and also adds a script to generate random normal variates. 

## Expectations

Reviews should seek to improve model fit.

This PR closes #3.

### CODE REVIEW

**Review by joshwlambert — CHANGES_REQUESTED** [review]
@jamesmbaazam Nice work on this PR, it definitely improves the package. 

Have you thought about using `drat` as a predictor in the regression to improve the fit.

File: R/get_norm_samples.R [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+#' @return A list of random normal samples with same length as the list elements
+#' @export
+#'
+#' @examples
+#' mylist <- list(a = 1, b = 1:10, c = 1:20)
+#' get_norm_samples(mylist)
+get_norm_samples <- function(mylist){
+  ll_rand <- list()
+  
+  for (i in 1:length(mylist)) {
```
Comment (joshwlambert) [inline R/get_norm_samples.R]:
```suggestion
  for (i in seq_along(mylist)) {
```

File: inst/model.R [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#' Model improvement
+
+# Data
+data <- mtcars
+
+# Fit a regression model to the data
+mod <- lm(mpg ~ wt + hp, data = data)
+
+# Print the model
+mod
```
Comment (joshwlambert) [inline inst/model.R]: Have you thought about plotting the results of the regression, using `plot()`?

File: R/get_norm_samples.R [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+#'
+#' @param mylist list of numeric vectors
+#' @return A list of random normal samples with same length as the list elements
+#' @export
+#'
+#' @examples
+#' mylist <- list(a = 1, b = 1:10, c = 1:20)
+#' get_norm_samples(mylist)
+get_norm_samples <- function(mylist){
+  ll_rand <- list()
```
Comment (joshwlambert) [inline R/get_norm_samples.R:12]:
```suggestion
# premake list
  ll_rand <- list()
```

File: R/get_norm_samples.R [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  
+  for (i in 1:length(mylist)) {
+    element_length <- length(mylist[[i]])
+    rnorm_samples <- rnorm(ll)
+    ll_rand[[i]] <- rnorm_samples
+    }
+  
+  return(ll_rand)
```
Comment (joshwlambert) [inline R/get_norm_samples.R:21]:
```suggestion
# return results
  return(ll_rand)
```

**Comment by jamesmbaazam** [comment]
> Have you thought about using `drat` as a predictor in the regression to improve the fit.

Oh, that's actually a good idea. I will do that in a separate issue.

File: inst/model.R (hunk 2 above)
Comment (jamesmbaazam) [reply inst/model.R:10]: Thanks for pointing that out. #6.

Comment (jamesmbaazam) [reply inst/model.R]: Thanks for pointing that out. I've made a change here [708bb98](https://github.com/jamesmbaazam/codereviewdemo/pull/4/commits/708bb98058d2de1019ccc8a96c7a2267bad9566e).

**Review by joshwlambert — APPROVED** [review]
Nice work with all the updates @jamesmbaazam 🚀

**Comment by jamesmbaazam** [comment]
Thanks for the review @joshwlambert

**Pull request merged**